test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -1033,7 +1033,7 @@ protected void wipeSearchableSnapshotsIndices() throws IOException {
                     adminClient().performRequest(new Request("DELETE", "/_snapshot/" + repoName + "/" + name));
                 }
             }
-            if (preserveReposUponCompletion() == false) {
+            if (preserveReposUponCompletion() == false && inProgressSnapshots.containsKey(repoName) == false) {
```

Comment:
I wonder if this is what we should do. The existing logic of deleting snapshots after test is already weak and I think this change will cause some repositories to be kept even if `preserveReposUponCompletion()` is set to `false.`

Instead I think we should change `wipeSnapshots()` (and its doc) so that it only deletes snapshots, and later in the calling method `wipeCluster()` we should check for `preserveReposUponCompletion()` and delete repositories there. We should always fail if there is an ongoing snapshot operation when deleting a repo and not silently ignoring it.